keymanapp/keyman
# fix(mac): avoid crash on startup with macOS 10.15 (Catalina)

To read the state and gain access to the Accessibility permission under the macOS Privacy & Security settings, Keyman calls the API `CGPreflightPostEventAccess`.  This API is [documented as being available with with macOS 10.15](https://developer.apple.com/documentation/coregraphics/3656520-cgpreflightposteventaccess?language=objc) (Catalina), but, based on experience, this does not appear to be the case. Limit use of API `CGPreflightPostEventAccess` to macOS 11.0 (Big Sur) and later, and, instead, use an older API, `AXIsProcessTrusted`, with OS versions before 11.0.

Fixes: #12295

# User Testing

## TEST_RUNS_SUCCESSFULLY_ON_MACOS_CATALINA

1. Install this version of Keyman on a system running macOS Catalina
1. Confirm that keyboards can be installed and that they appear in the Keyman menu
1. Type with the installed Keyman keyboards and confirm that keystrokes are processed as expected
